KOGITO-5508 Data Index Dev Service (#3991)
kiegroup/kie-docs: doc-content/kogito-docs/src/main/asciidoc/configuration/chap-kogito-configuring.adoc

NOTE: Data Index Service with PostgreSQL does not support indexing domain-specific data. The GraphQL schema only exposes queries for `ProcessInstances`, `UserTaskInstances`, and `Jobs`.

[id="proc-data-index-dev-service_{context}"]
=== {PRODUCT} Quarkus Dev Service for Data Index

When using the {PRODUCT} Process Quarkus extension, an ephemeral Data Index service is automatically provisoned when running Quarkus on Development mode.

This means that once you start your Quarkus project on dev mode, an in-memory instance of the Data Index service is automatically
started in the background. This feature is enabled by https://quarkus.io/guides/dev-services[Quarkus Dev Services], and leverages on https://www.testcontainers.org/[Testcontainers] to start an image of the Data Index service.

> The {PRODUCT} Process Quarkus extension will also set up your Quarkus application to automatically replicate any

Reviewer: @cristianonicolai  maybe we can add here some kind of note or reference  about how to add that extension to a kogito project .I mean the necessary dependency or a link to how the extension is used. Wdty?

Author: +1, added reference to the extensions dependency

NOTE: Data Index Service with PostgreSQL does not support indexing domain-specific data. The GraphQL schema only exposes queries for `ProcessInstances`, `UserTaskInstances`, and `Jobs`.

[id="proc-data-index-dev-service_{context}"]
=== {PRODUCT} Quarkus Dev Service for Data Index

When using the {PRODUCT} Process Quarkus extension, an ephemeral Data Index service is automatically provisoned when running Quarkus on Development mode. Whenver you're using either one of the following Quarkus extensions, Dev Service will be setup out-of-the-box:

[source]
----
<dependency>
  <groupId>org.kie.kogito</groupId>
  <artifactId>kogito-quarkus</artifactId>
</dependency>
----
or
[source]
----
<dependency>
  <groupId>org.kie.kogito</groupId>
  <artifactId>kogito-quarkus-processes</artifactId>
</dependency>
----

This means that once you start your Quarkus project on dev mode, an in-memory instance of the Data Index service is automatically
started in the background. This feature is enabled by https://quarkus.io/guides/dev-services[Quarkus Dev Services], and leverages on https://www.testcontainers.org/[Testcontainers] to start an image of the Data Index service.

> The {PRODUCT} Process Quarkus extension will also set up your Quarkus application to automatically replicate any

Reviewer 2: ```suggestion
The {PRODUCT} Process Quarkus extension sets up your Quarkus application to automatically replicate any
```
